gdal/frmts/fits/fitsdataset.cpp

```diff
@@ -599,6 +720,684 @@ GDALDataset *FITSDataset::Create( const char* pszFilename,
+            }
+        }
+
+        CPLFree( pszProjection );
+        pszProjection = nullptr;
+
+        /* Waiting for linear keywords standardization only deg ctype are used */
+        /* Check if WCS are there */
+        fits_read_key(hFITS, TSTRING, "CTYPE1", ctype, nullptr, &status);
+        if ( !status ) {
+            /* Check if angular WCS are there */
+            if ( strstr(ctype, "LN") )
+            {
+                /* Reading reference points */
+                fits_read_key(hFITS, TDOUBLE, "CRPIX1", &crpix1, nullptr, &status);
+                fits_read_key(hFITS, TDOUBLE, "CRPIX2", &crpix2, nullptr, &status);
+                fits_read_key(hFITS, TDOUBLE, "CRVAL1", &crval1, nullptr, &status);
+                fits_read_key(hFITS, TDOUBLE, "CRVAL2", &crval2, nullptr, &status);
+                /* Check for CDELT and PC matrix representation */
+                fits_read_key(hFITS, TDOUBLE, "CDELT1", &cdelt1, nullptr, &status);
+                if ( ! status ) {
+                    fits_read_key(hFITS, TDOUBLE, "CDELT2", &cdelt2, nullptr, &status);
+                    fits_read_key(hFITS, TDOUBLE, "PC1_1", &pc[0], nullptr, &status);
+                    fits_read_key(hFITS, TDOUBLE, "PC1_2", &pc[1], nullptr, &status);
+                    fits_read_key(hFITS, TDOUBLE, "PC2_1", &pc[2], nullptr, &status);
+                    fits_read_key(hFITS, TDOUBLE, "PC2_2", &pc[3], nullptr, &status);
+                    cd[0] = cdelt1 * pc[0];
+                    cd[1] = cdelt1 * pc[1];
+                    cd[2] = cdelt2 * pc[2];
+                    cd[3] = cdelt2 * pc[3];
+                    status = 0;
+                } else {
+                    /* Look for CD matrix representation */
+                    fits_read_key(hFITS, TDOUBLE, "CD1_1", &cd[0], nullptr, &status);
+                    fits_read_key(hFITS, TDOUBLE, "CD1_2", &cd[1], nullptr, &status);
+                    fits_read_key(hFITS, TDOUBLE, "CD2_1", &cd[2], nullptr, &status);
+                    fits_read_key(hFITS, TDOUBLE, "CD2_2", &cd[3], nullptr, &status);
+                }
+
+                double radfac = DEG2RAD * aRadius;
+
+                adfGeoTransform[1] = cd[0] * radfac;
+                adfGeoTransform[2] = cd[1] * radfac;
+                adfGeoTransform[4] = cd[2] * radfac;
+                adfGeoTransform[5] = - cd[3] * radfac ;
+                if ( crval1 > 180. ) {
+                    crval1 = crval1 - 180.;
+                }
+                adfGeoTransform[0] = crval1 * radfac - adfGeoTransform[1] *
+                                                    (crpix1-0.5);
+                adfGeoTransform[3] = crval2 * radfac - adfGeoTransform[5] *
+                                                    (crpix2-0.5);
+                bGeoTransformValid = true;
+            }
+
+            char* pstr = strrchr(ctype, '-') + 1;
+            char projstr[4];
+            strcpy(projstr, pstr);
```

Comment:
this could overflow projstr[4]. Do you really need to make a copy ?

Reply:
Nope, sorry, fixed in 41699ae24b983381a1a63715fbbae71c442059d8